frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.ts

```diff
@@ -15,106 +20,178 @@ import {
+        const label = artifactKey;
+        const { artifactId } =
+          artifactNodeData?.find((a) => artifactKey === a.outputArtifactKey) ?? {};
+
+        // if no node needs it as an input, we don't really need a well known id
+        const id = artifactId ?? artifactKey;
+
+        nodes.push(
+          createArtifactNode({
+            id,
+            label,
+            artifactType: data.schemaTitle,
+            runAfter: [taskId],
+            status: translateStatusForNode(status?.state),
+          }),
         );
-      } else if (parentTask) {
-        // Create an edge from the grouped task to its parent task
-        // Prevent the node floating on the topology
-        // This logic could be removed once we have the stacked node to better deal with groups
-        runAfter.push(parentTask);
-      }
 
+        newTaskMapEntries[id] = {
+          type: 'artifact',
+          name: label,
+          inputs: {
+            artifacts: [{ label: id, type: composeArtifactType(data) }],
+          },
+        };
+      });
+    }
+
+    // This task
+    newTaskMapEntries[taskId] = {
+      type: isGroupNode ? 'groupTask' : 'task',
+      name: taskName,
+      steps: executor ? [executor.container] : undefined,
+      inputs: parseInputOutput(component?.inputDefinitions),
+      outputs: parseInputOutput(component?.outputDefinitions),
+      status,
+      volumeMounts: parseVolumeMounts(spec.platform_spec, executorLabel),
+    };
+    if (taskValue.dependentTasks) {
+      // This task's runAfters may need artifact relationships -- find those artifactIds
+      runAfter.push(
+        ...taskValue.dependentTasks
+          .map((dependantTaskId) => {
+            const art = taskArtifactMap[dependantTaskId];
+            return art ? art.map((v) => v.artifactId) : null;
+          })
+          .filter((v): v is string[] => !!v)
+          .flat(),
+      );
+    }
+
+    // This task's rendering information
+    if (isGroupNode && groupTasks) {
+      const [nestedNodes, children] = getNestedNodes(groupTasks, components, runDetails);
+      const itemNode = createGroupNode(
+        {
+          id: taskId,
+          label: taskName,
+          runAfter,
+          status: translateStatusForNode(status?.state),
+        },
+        children,
+      );
+      nodes.push(itemNode, ...nestedNodes);
+
+      // Extract IDs and create new entries
+      nestedNodes.forEach((node) => {
+        const { id } = node;
+
+        newTaskMapEntries[id] = {
+          type: 'groupTask',
+          name: id,
+          steps: executor ? [executor.container] : undefined,
+          // TODO: render/parse sidebar data
+          // inputs: parseInputOutput(component.dag.tasks[id]?.inputs),
+          // outputs: parseInputOutput(component.dag.tasks[id]?.inputs),
+          status,
+          volumeMounts: parseVolumeMounts(spec.platform_spec, executorLabel),
+        };
```

Comment:
Adding the `PipelineTask` as a field on the node's data would remove the need for the `TaskMap` and removing this extra mapping of nodes to task data. When the nodes are created in `getNestedNodes` the inputs and outputs could be set (since in knows the component) and remove this `TODO`.